### DISCUSSION TITLE
How can I use useHover() with a functional component?

### DISCUSSION THREAD

**Discussion by augustomallmann** [floating-ui/floating-ui#1830] [React]
Hey,

I'm having some problems creating a reusable tooltip component. The user should be able to apply it to react functional components imported from a design system and not only plain text / html text. 

**What works:** when the user uses plain text or a html component, the useHover() function identifies the mouse hover and display the tooltip.

**Problem:**  when using a react functional component, the hover event is not detected and nothing happens on the ui. In the console, there's the following warning:

```
Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?
```

I created a codesandbox with the code: https://codesandbox.io/s/charming-wilson-hhb9m4?file=/src/Tooltip.tsx

The blue button is imported from the design system and don't works on hover. So far, I identified that it is related the `cloneElement` function.

**Comment by atomiks** [comment]
The playbook ui library needs to be forwarding refs (as the console suggests) in order for refs to be attached to the actual DOM node for measurement by this library. https://reactjs.org/docs/forwarding-refs.html

**Reply by augustomallmann** [reply]
yes, but as the package doesn't do that, is there a workaround?

**Reply by FezVrasta** [reply]
It's the component you want to use as child that needs to properly forward the `ref`

**Reply by augustomallmann** [reply]
yes but it comes from the `playbook-ui` library, I don't have access to that

**Reply by FezVrasta** [reply]
I understand, but in order to perform the logic required by Floating UI to position your elements we need access to an HTML element ref. I never used playbook-ui but usually these libraries do provide a way to access the inner ref pointer.

In alternative you may wrap the component in a `div` or other suitable HTML element and position the tooltip relative to it.
